src/docs/arch-dhandle.dox

```diff
@@ -1,17 +1,131 @@
+The closure of the dhandle often occurs on the next iteration of the 
+sweep when the sweep server detects the dhandle has remained idle for 
+long enough. The server marks the dhandle as dead so that the next time 
+a session with a reference walks its own cache list, it will see the 
+handle marked dead and remove it from the session's dhandle cache list.
+
+The sweep server then checks for any sessions that are referencing this 
+dhandle. If a dhandle stays referenced by at least one session 
+(\c session_ref count > 0), the dhandle cannot be removed from the 
+global list. If the dhandle is not referenced by any session, the sweep 
+server removes the dhandle from the global dhandle list and frees any 
+remaining resources associated with it. The removal of the dhandle from 
+the global list completes the lifecycle, and any future access of the 
+associated table will require the dhandle to be created again.
+
+@section dhandle_dhandle_session_cache_sweep
+
+A session's dhandle cache list is periodically cleaned out through a 
+dhandle cache sweep, which removes references to the dhandles that 
+have not been accessed by the session in a long time. Sessions have 
+their own sweeps; since a session is single-threaded, a session's 
+dhandle cache can only be altered by that session alone.
+
+Each time a session accesses a dhandle, it checks if enough time has 
+elapsed to do a session cache sweep for that session. As it walks the 
+session dhandle cache list, it notices if any dhandle on its list has 
+been marked dead (idle for too long). If it has, the session removes 
+that dhandle from its list and decrements the session_ref count.
+
+@section btree_btrees B-Trees
+
+Database tables are commonly represented by a B-Tree data structure, which 
+is made up of nodes that are page structures. The root and internal pages will 
+only store keys and references to other pages, while leaf pages store keys, 
+values and sometimes references to overflow pages. The focus here will be on 
+the in-memory representation of the B-Tree which is defined in \c btmem.h. The 
+on-disk representation and organization of pages is discussed in the 
+@ref arch-data-file page.
+
+@section btree_btree_data_handle B-Tree Data Handle (WT_BTREE)
+
+The B-Tree data handle serves as a way to access the underlying B-Tree data 
+structure. The \c WT_BTREE structure (defined in \c btree.h) contains a 
+reference to its corresponding generic data handle, checkpoint information, the 
+specific WT_BTREE type, key and value formats, collator, key and page size 
+configurations, split information, a reference to the root page on disk, write 
+generation information, sync information, eviction information, and a set of flags.
+
+@section btree_btree_in_memory_representation B-Tree In-Memory Representation
+
+Inside a \c WT_BTREE, the reference to the root page is a \c WT_REF, which 
+represents a single in-memory page and its state information. The \c WT_REF 
```

Comment:
I don't think a "represents a single in-memory page" is correct, in particular "in-memory".  The `WT_REF` is a reference to a page that may or may not be loaded from storage.  Once the page is loaded, the `WT_REF` will have a valid `WT_PAGE` pointer (and it does represent the in-memory page).